### ISSUE TITLE
`--with-sqlite` build error

### ISSUE THREAD

**Issue by digizeph** [CAIDA/libbgpstream#157]
- Configured via `./configure --with-sqlite`.
- MacOS, with sqlite intalled via homebrew.
- Build error: `bsdi_sqlite.c:345:5: error: unknown type name 'bgpstream_record_dump_type_t'; did you mean 'bgpstream_record_type_t'?`

Full log:
```
mingwei@pug:~/git/caida-git/libbgpstream$ make
/Library/Developer/CommandLineTools/usr/bin/make  all-recursive
Making all in common
Making all in libpatricia
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..     -g -O2 -D_REENTRANT  -MT patricia.lo -MD -MP -MF .deps/patricia.Tpo -c -o patricia.lo patricia.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -g -O2 -D_REENTRANT -MT patricia.lo -MD -MP -MF .deps/patricia.Tpo -c patricia.c  -fno-common -DPIC -o .libs/patricia.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -g -O2 -D_REENTRANT -MT patricia.lo -MD -MP -MF .deps/patricia.Tpo -c patricia.c -o patricia.o >/dev/null 2>&1
mv -f .deps/patricia.Tpo .deps/patricia.Plo
/bin/sh ../../libtool  --tag=CC   --mode=link gcc  -g -O2 -D_REENTRANT    -o libpatricia.la  patricia.lo  -lsqlite3 -lrdkafka -lwandio
libtool: link: ar cru .libs/libpatricia.a .libs/patricia.o
libtool: link: ranlib .libs/libpatricia.a
libtool: link: ( cd ".libs" && rm -f "libpatricia.la" && ln -s "../libpatricia.la" "libpatricia.la" )
Making all in libinterval3
Making all in rb_tree
/bin/sh ../../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../..     -g -O2 -D_REENTRANT  -MT stack.lo -MD -MP -MF .deps/stack.Tpo -c -o stack.lo stack.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -g -O2 -D_REENTRANT -MT stack.lo -MD -MP -MF .deps/stack.Tpo -c stack.c  -fno-common -DPIC -o .libs/stack.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -g -O2 -D_REENTRANT -MT stack.lo -MD -MP -MF .deps/stack.Tpo -c stack.c -o stack.o >/dev/null 2>&1
mv -f .deps/stack.Tpo .deps/stack.Plo
/bin/sh ../../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../..     -g -O2 -D_REENTRANT  -MT misc.lo -MD -MP -MF .deps/misc.Tpo -c -o misc.lo misc.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -g -O2 -D_REENTRANT -MT misc.lo -MD -MP -MF .deps/misc.Tpo -c misc.c  -fno-common -DPIC -o .libs/misc.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -g -O2 -D_REENTRANT -MT misc.lo -MD -MP -MF .deps/misc.Tpo -c misc.c -o misc.o >/dev/null 2>&1
mv -f .deps/misc.Tpo .deps/misc.Plo
/bin/sh ../../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../..     -g -O2 -D_REENTRANT  -MT red_black_tree.lo -MD -MP -MF .deps/red_black_tree.Tpo -c -o red_black_tree.lo red_black_tree.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -g -O2 -D_REENTRANT -MT red_black_tree.lo -MD -MP -MF .deps/red_black_tree.Tpo -c red_black_tree.c  -fno-common -DPIC -o .libs/red_black_tree.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../.. -g -O2 -D_REENTRANT -MT red_black_tree.lo -MD -MP -MF .deps/red_black_tree.Tpo -c red_black_tree.c -o red_black_tree.o >/dev/null 2>&1
mv -f .deps/red_black_tree.Tpo .deps/red_black_tree.Plo
/bin/sh ../../../libtool  --tag=CC   --mode=link gcc  -g -O2 -D_REENTRANT    -o librbtree.la  stack.lo misc.lo red_black_tree.lo  -lsqlite3 -lrdkafka -lwandio
libtool: link: ar cru .libs/librbtree.a .libs/stack.o .libs/misc.o .libs/red_black_tree.o
libtool: link: ranlib .libs/librbtree.a
libtool: link: ( cd ".libs" && rm -f "librbtree.la" && ln -s "../librbtree.la" "librbtree.la" )
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../../common/libinterval3/rb_tree   -g -O2 -D_REENTRANT  -MT interval_tree.lo -MD -MP -MF .deps/interval_tree.Tpo -c -o interval_tree.lo interval_tree.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../../common/libinterval3/rb_tree -g -O2 -D_REENTRANT -MT interval_tree.lo -MD -MP -MF .deps/interval_tree.Tpo -c interval_tree.c  -fno-common -DPIC -o .libs/interval_tree.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../../common/libinterval3/rb_tree -g -O2 -D_REENTRANT -MT interval_tree.lo -MD -MP -MF .deps/interval_tree.Tpo -c interval_tree.c -o interval_tree.o >/dev/null 2>&1
mv -f .deps/interval_tree.Tpo .deps/interval_tree.Plo
/bin/sh ../../libtool  --tag=CC   --mode=link gcc  -g -O2 -D_REENTRANT    -o libinterval3.la  interval_tree.lo ../../common/libinterval3/rb_tree/librbtree.la  -lsqlite3 -lrdkafka -lwandio
libtool: link: (cd .libs/libinterval3.lax/librbtree.a && ar x "/Users/mingwei/git/caida-git/libbgpstream/common/libinterval3/../../common/libinterval3/rb_tree/.libs/librbtree.a")
libtool: link: ar cru .libs/libinterval3.a .libs/interval_tree.o   .libs/libinterval3.lax/librbtree.a/misc.o .libs/libinterval3.lax/librbtree.a/red_black_tree.o .libs/libinterval3.lax/librbtree.a/stack.o
libtool: link: ranlib .libs/libinterval3.a
libtool: link: rm -fr .libs/libinterval3.lax
libtool: link: ( cd ".libs" && rm -f "libinterval3.la" && ln -s "../libinterval3.la" "libinterval3.la" )
Making all in libcsv
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..     -g -O2 -D_REENTRANT  -MT libcsv.lo -MD -MP -MF .deps/libcsv.Tpo -c -o libcsv.lo libcsv.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -g -O2 -D_REENTRANT -MT libcsv.lo -MD -MP -MF .deps/libcsv.Tpo -c libcsv.c  -fno-common -DPIC -o .libs/libcsv.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -g -O2 -D_REENTRANT -MT libcsv.lo -MD -MP -MF .deps/libcsv.Tpo -c libcsv.c -o libcsv.o >/dev/null 2>&1
mv -f .deps/libcsv.Tpo .deps/libcsv.Plo
/bin/sh ../../libtool  --tag=CC   --mode=link gcc  -g -O2 -D_REENTRANT    -o libcsv.la  libcsv.lo  -lsqlite3 -lrdkafka -lwandio
libtool: link: ar cru .libs/libcsv.a .libs/libcsv.o
libtool: link: ranlib .libs/libcsv.a
libtool: link: ( cd ".libs" && rm -f "libcsv.la" && ln -s "../libcsv.la" "libcsv.la" )
Making all in libjsmn
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..     -g -O2 -D_REENTRANT  -MT jsmn.lo -MD -MP -MF .deps/jsmn.Tpo -c -o jsmn.lo jsmn.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -g -O2 -D_REENTRANT -MT jsmn.lo -MD -MP -MF .deps/jsmn.Tpo -c jsmn.c  -fno-common -DPIC -o .libs/jsmn.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -g -O2 -D_REENTRANT -MT jsmn.lo -MD -MP -MF .deps/jsmn.Tpo -c jsmn.c -o jsmn.o >/dev/null 2>&1
mv -f .deps/jsmn.Tpo .deps/jsmn.Plo
/bin/sh ../../libtool  --tag=CC   --mode=link gcc  -g -O2 -D_REENTRANT    -o libjsmn.la  jsmn.lo  -lsqlite3 -lrdkafka -lwandio
libtool: link: ar cru .libs/libjsmn.a .libs/jsmn.o
libtool: link: ranlib .libs/libjsmn.a
libtool: link: ( cd ".libs" && rm -f "libjsmn.la" && ln -s "../libjsmn.la" "libjsmn.la" )
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn   -g -O2 -D_REENTRANT  -MT utils.lo -MD -MP -MF .deps/utils.Tpo -c -o utils.lo utils.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT utils.lo -MD -MP -MF .deps/utils.Tpo -c utils.c  -fno-common -DPIC -o .libs/utils.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT utils.lo -MD -MP -MF .deps/utils.Tpo -c utils.c -o utils.o >/dev/null 2>&1
mv -f .deps/utils.Tpo .deps/utils.Plo
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn   -g -O2 -D_REENTRANT  -MT ip_utils.lo -MD -MP -MF .deps/ip_utils.Tpo -c -o ip_utils.lo ip_utils.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT ip_utils.lo -MD -MP -MF .deps/ip_utils.Tpo -c ip_utils.c  -fno-common -DPIC -o .libs/ip_utils.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT ip_utils.lo -MD -MP -MF .deps/ip_utils.Tpo -c ip_utils.c -o ip_utils.o >/dev/null 2>&1
mv -f .deps/ip_utils.Tpo .deps/ip_utils.Plo
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn   -g -O2 -D_REENTRANT  -MT jsmn_utils.lo -MD -MP -MF .deps/jsmn_utils.Tpo -c -o jsmn_utils.lo jsmn_utils.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT jsmn_utils.lo -MD -MP -MF .deps/jsmn_utils.Tpo -c jsmn_utils.c  -fno-common -DPIC -o .libs/jsmn_utils.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT jsmn_utils.lo -MD -MP -MF .deps/jsmn_utils.Tpo -c jsmn_utils.c -o jsmn_utils.o >/dev/null 2>&1
mv -f .deps/jsmn_utils.Tpo .deps/jsmn_utils.Plo
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..  -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn   -g -O2 -D_REENTRANT  -MT parse_cmd.lo -MD -MP -MF .deps/parse_cmd.Tpo -c -o parse_cmd.lo parse_cmd.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT parse_cmd.lo -MD -MP -MF .deps/parse_cmd.Tpo -c parse_cmd.c  -fno-common -DPIC -o .libs/parse_cmd.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I.. -I../common/libpatricia -I../common/libpinterval3 -I../common/libcsv -I../common/libjsmn -g -O2 -D_REENTRANT -MT parse_cmd.lo -MD -MP -MF .deps/parse_cmd.Tpo -c parse_cmd.c -o parse_cmd.o >/dev/null 2>&1
mv -f .deps/parse_cmd.Tpo .deps/parse_cmd.Plo
/bin/sh ../libtool  --tag=CC   --mode=link gcc  -g -O2 -D_REENTRANT    -o libcccommon.la  utils.lo ip_utils.lo jsmn_utils.lo parse_cmd.lo ../common/libpatricia/libpatricia.la ../common/libinterval3/libinterval3.la ../common/libcsv/libcsv.la ../common/libjsmn/libjsmn.la  -lsqlite3 -lrdkafka -lwandio
libtool: link: (cd .libs/libcccommon.lax/libpatricia.a && ar x "/Users/mingwei/git/caida-git/libbgpstream/common/../common/libpatricia/.libs/libpatricia.a")
libtool: link: (cd .libs/libcccommon.lax/libinterval3.a && ar x "/Users/mingwei/git/caida-git/libbgpstream/common/../common/libinterval3/.libs/libinterval3.a")
libtool: link: (cd .libs/libcccommon.lax/libcsv.a && ar x "/Users/mingwei/git/caida-git/libbgpstream/common/../common/libcsv/.libs/libcsv.a")
libtool: link: (cd .libs/libcccommon.lax/libjsmn.a && ar x "/Users/mingwei/git/caida-git/libbgpstream/common/../common/libjsmn/.libs/libjsmn.a")
libtool: link: ar cru .libs/libcccommon.a .libs/utils.o .libs/ip_utils.o .libs/jsmn_utils.o .libs/parse_cmd.o   .libs/libcccommon.lax/libpatricia.a/patricia.o  .libs/libcccommon.lax/libinterval3.a/interval_tree.o .libs/libcccommon.lax/libinterval3.a/misc.o .libs/libcccommon.lax/libinterval3.a/red_black_tree.o .libs/libcccommon.lax/libinterval3.a/stack.o  .libs/libcccommon.lax/libcsv.a/libcsv.o  .libs/libcccommon.lax/libjsmn.a/jsmn.o
libtool: link: ranlib .libs/libcccommon.a
libtool: link: rm -fr .libs/libcccommon.lax
libtool: link: ( cd ".libs" && rm -f "libcccommon.la" && ln -s "../libcccommon.la" "libcccommon.la" )
Making all in lib
Making all in datainterfaces
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../.. -I../../lib -I../../lib/utils -I../../common   -g -O2 -D_REENTRANT  -MT bsdi_broker.lo -MD -MP -MF .deps/bsdi_broker.Tpo -c -o bsdi_broker.lo bsdi_broker.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_broker.lo -MD -MP -MF .deps/bsdi_broker.Tpo -c bsdi_broker.c  -fno-common -DPIC -o .libs/bsdi_broker.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_broker.lo -MD -MP -MF .deps/bsdi_broker.Tpo -c bsdi_broker.c -o bsdi_broker.o >/dev/null 2>&1
mv -f .deps/bsdi_broker.Tpo .deps/bsdi_broker.Plo
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../.. -I../../lib -I../../lib/utils -I../../common   -g -O2 -D_REENTRANT  -MT bsdi_singlefile.lo -MD -MP -MF .deps/bsdi_singlefile.Tpo -c -o bsdi_singlefile.lo bsdi_singlefile.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_singlefile.lo -MD -MP -MF .deps/bsdi_singlefile.Tpo -c bsdi_singlefile.c  -fno-common -DPIC -o .libs/bsdi_singlefile.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_singlefile.lo -MD -MP -MF .deps/bsdi_singlefile.Tpo -c bsdi_singlefile.c -o bsdi_singlefile.o >/dev/null 2>&1
mv -f .deps/bsdi_singlefile.Tpo .deps/bsdi_singlefile.Plo
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../.. -I../../lib -I../../lib/utils -I../../common   -g -O2 -D_REENTRANT  -MT bsdi_kafka.lo -MD -MP -MF .deps/bsdi_kafka.Tpo -c -o bsdi_kafka.lo bsdi_kafka.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_kafka.lo -MD -MP -MF .deps/bsdi_kafka.Tpo -c bsdi_kafka.c  -fno-common -DPIC -o .libs/bsdi_kafka.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_kafka.lo -MD -MP -MF .deps/bsdi_kafka.Tpo -c bsdi_kafka.c -o bsdi_kafka.o >/dev/null 2>&1
mv -f .deps/bsdi_kafka.Tpo .deps/bsdi_kafka.Plo
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../.. -I../../lib -I../../lib/utils -I../../common   -g -O2 -D_REENTRANT  -MT bsdi_csvfile.lo -MD -MP -MF .deps/bsdi_csvfile.Tpo -c -o bsdi_csvfile.lo bsdi_csvfile.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_csvfile.lo -MD -MP -MF .deps/bsdi_csvfile.Tpo -c bsdi_csvfile.c  -fno-common -DPIC -o .libs/bsdi_csvfile.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_csvfile.lo -MD -MP -MF .deps/bsdi_csvfile.Tpo -c bsdi_csvfile.c -o bsdi_csvfile.o >/dev/null 2>&1
mv -f .deps/bsdi_csvfile.Tpo .deps/bsdi_csvfile.Plo
/bin/sh ../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../..  -I../.. -I../../lib -I../../lib/utils -I../../common   -g -O2 -D_REENTRANT  -MT bsdi_sqlite.lo -MD -MP -MF .deps/bsdi_sqlite.Tpo -c -o bsdi_sqlite.lo bsdi_sqlite.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../.. -I../../lib -I../../lib/utils -I../../common -g -O2 -D_REENTRANT -MT bsdi_sqlite.lo -MD -MP -MF .deps/bsdi_sqlite.Tpo -c bsdi_sqlite.c  -fno-common -DPIC -o .libs/bsdi_sqlite.o
bsdi_sqlite.c:345:5: error: unknown type name 'bgpstream_record_dump_type_t'; did you mean 'bgpstream_record_type_t'?
    bgpstream_record_dump_type_t type;
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
    bgpstream_record_type_t
../../lib/bgpstream_record.h:76:3: note: 'bgpstream_record_type_t' declared here
} bgpstream_record_type_t;
  ^
1 error generated.
make[4]: *** [bsdi_sqlite.lo] Error 1
make[3]: *** [all-recursive] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

**Comment by alistairking** [comment]
Oh right. We probably haven't updated the SQLite data interface since we reworked the core of the library.

**Comment by digizeph** [comment]
Fixed in PR #158
